### ISSUE TITLE
astropy.convolution.convolve won't accept pandas.Series as input

### ISSUE THREAD

**Issue by janerigby** [astropy/astropy#6099]
The issue:  Pandas Series can be passed to numpy functions just as if they were a numpy array.  However, astropy.convolution.convolve won't accept them.

The workaround: pass the Pandas series to convolve() with .as_matrix(), so that it is sent as a numpy array.  See example below.

The proposed change to Astropy:    convolve() should accept a Pandas Series as input.

Example code:
```python
import astropy.convolution
import numpy as np
import pandas
wave = np.array((np.linspace(5000,5100,10)))
wave_series = pandas.Series(wave)
boxcar = 3
this_works = astropy.convolution.convolve(wave, np.ones((boxcar,))/boxcar)
this_fails  = astropy.convolution.convolve(wave_series) np.ones((boxcar,))/boxcar)
workaround  = astropy.convolution.convolve(wave_series.as_matrix(), np.ones((boxcar,))/boxcar)
```

**Comment by cdeil** [comment]
@janerigby - There's probably 100s of functions / methods in Astropy that don't work with pandas, only with numpy arrays. Adding support for pandas in a meaningful way (i.e. not just one random function, but not others) would be a big task (many code changes, test additions, docs changes).

So given that there's a straightforward way to interoperate (pandas objects expose the data as numpy arrays that can be passed to Astropy), maybe it's best to not go in that direction of adding pandas support to Astropy?

There is http://docs.astropy.org/en/stable/table/pandas.html mentioning the relation and interoperation of Astropy Table and pandas DataFrame (as far as I know this Table <-> DataFrame translation is the only direct pandas support that was added to Astropy. note that it's localised in one place and thus wasn't hard to add / test / document / maintain). Probably another docs section (not sure where) mentioning how to work with Astropy in general and pandas would be a good addition (using e.g. your convolution example as one of the examples shown)

**Comment by astrofrog** [comment]
It might be sufficient to use ``np.asarray`` in astropy functions that can accept arrays or array-like objects without adding explicit support for pandas:

```
In [8]: np.asarray(wave_series)
Out[8]: 
array([ 5000.        ,  5011.11111111,  5022.22222222,  5033.33333333,
        5044.44444444,  5055.55555556,  5066.66666667,  5077.77777778,
        5088.88888889,  5100.        ])
```

**Comment by keflavich** [comment]
In principle, this should have been solved by #6267 / #5782.  I think we just need to add a test to close this issue?

**Comment by bsipocz** [comment]
@keflavich - sounds like a good solution. Added this to the next milestone, so we won't forget about it.
